packages/shopify/react/components/form.tsx

```diff
@@ -0,0 +1,146 @@
+    formAttributes.action = '/account/activate';
+  } else if (type === 'contact') {
+    formAttributes.action = '/contact#contact_form';
+    formAttributes.id = 'contact_form';
+    formAttributes.className = 'contact-form';
+  } else if (type === 'currency') {
+    formAttributes.action = '/cart/update';
+    formAttributes.id = 'currency_form';
+    formAttributes.encType = 'multipart/form-data';
+    formAttributes.className = 'shopify-currency-form';
+  } else if (type === 'customer') {
+    formAttributes.action = '/contact#contact_form';
+    formAttributes.id = 'contact_form';
+    formAttributes.className = 'contact-form';
+  } else if (type === 'create_customer') {
+    formAttributes.action = '/account';
+    formAttributes.id = 'create_customer';
+  } else if (type === 'customer_address') {
+    formAttributes.action = '/account/addresses';
+    formAttributes.id = 'address_form_new';
+  } else if (type === 'customer_login') {
+    formAttributes.action = '/account/login';
+  } else if (type === 'guest_login') {
+    formAttributes.action = '/account/login';
+    formAttributes.id = 'customer_login_guest';
+  } else if (type === 'product') {
+    const product = builderState?.context.shopify.liquid.get(parameter, builderState.state);
+    formAttributes.action = '/cart/add';
+    formAttributes.id = `product_form_${product?.id}`;
+    formAttributes.className = 'shopify-product-form';
+    formAttributes.encType = 'multipart/form-data';
+  } else if (type === 'recover_customer_password') {
+    formAttributes.action = '/account/recover';
+  } else if (type === 'reset_customer_password') {
+    formAttributes.action = '/account/reset';
+  } else if (type === 'storefront_password') {
+    formAttributes.action = '/password';
+    formAttributes.id = 'login_form';
+    formAttributes.className = 'storefront-password-form';
+  } else if (type === 'new_comment') {
+    formAttributes.action = '/blogs/news/my-blog-post/comments#comment_form';
+    formAttributes.id = 'comment_form';
+    formAttributes.className = 'comment-form';
+  }
+
+  if (customAttributes?.length) {
+    customAttributes.forEach(attribute => {
+      const [key, value] = attribute.split(':');
+      let useValue = value?.trim();
+
+      if (useValue && builderState?.state) {
+        useValue = builderState.context.shopify.liquid.render(
+          `{{${useValue}}}`,
+          builderState.state
+        );
+      }
+
+      if (key === 'class' && formAttributes.className && useValue) {
+        formAttributes.className = `${formAttributes.className} ${useValue?.trim() || ''}`;
+      } else if (useValue) {
+        formAttributes[key] = useValue;
+      } else if (key && !useValue) {
+        // This is an attribute that does not have a value but should be added to the attribute list (e.g. boolean attribute)
+        formAttributes[key] = true;
+      }
+    });
+  }
+
+  return (
+    <BuilderStoreContext.Consumer>
+      {store => (
```

Comment:
Using 
```tsx
<StateProvider state={{form}} > <...></StateProvider>
```
should do the same as this

Reply:
Oh, does it work with jsx like the `form` stuff I am doing? Looking at the component I thought it takes a `builderBlock` that has children as a prop and then renders them, but not children elements:
```js
class StateProviderComponent extends React.Component<StateProviderProps> {
  render() {
    return (
      <BuilderStoreContext.Consumer>
        {state => (
          <BuilderStoreContext.Provider
            value={{
              ...state,
              state: {
                ...state.state,
                ...this.props.state
              },
              context: {
                ...state.context,
                ...this.props.context
              }
            }}
          >
            {this.props.builderBlock &&
              this.props.builderBlock.children &&
              this.props.builderBlock.children.map((block, index) => (
                <BuilderBlockComponent
                  block={block}
                  key={block.id}
                  index={index}
                  child={true} 
                />
              ))}
            {this.props.children}
          </BuilderStoreContext.Provider>
        )}
      </BuilderStoreContext.Consumer>
    )
  }
}
```

Reply:
it doesn't look like it, what you have should work fine